### ISSUE TITLE
BufferOverflowException with things growing over size limit

### ISSUE THREAD

**Issue by tobias-zeptio** [eclipse-ditto/ditto#1499]
We have the size limit variable set to `LIMITS_THINGS_MAX_SIZE=100k`, but still things_snaps contain document above that limit. The connectivity service logs do have a ThingTooLargeException. We have a payload-mapper calling a merge command for each update.

One example I ran `bsonsize(db.things_snaps.findOne({ pid: "thing:twin:4e36a386-697e-4be6-bf1c-8d1b32c837db" }))` and got 892698.

This is a deployment running from ditto 2.4, and yesterday updated to 3.0.0.

 One theory is that multiple mapper called happen in parallel, causing the size limit to accumulate before the size has been updated to the mapper.

**Comment by thjaeckle** [comment]
@tobias-zeptio So you created a Thing with Ditto 2.4 exceeding the limit (e.g. because of the in 2.4 missing "size enforcement" for metadata) and this is still "too big" in Ditto 3.0?

Can you reproduce that (creating a "too big thing") with Ditto 3.0?
If not, I suggest you delete the thing from persistence and recreate it.

**Comment by tobias-zeptio** [comment]
This wasn't because of _metadata, that was another issue. ;) I've already delete the things properties having this issue, since the size limit was preventing updates. So the size limit worked when the thing was large, but didn't prevent it from getting to large-

In this case is was features/properties growing. I'll try to reproduce using test data.

**Comment by thjaeckle** [comment]
Please open a new issue if you are able to reproduce.
